packages/storage/src/components/ImageGallery/image.js

```diff
@@ -28,16 +34,52 @@ export default function Image(props) {
         })();
     }, [imageRef, onError]);
 
+    const handleClick = (e) => {
+        const isClickDeleteBtn = deleteBtn.current.htmlElementRef.current.contains(e.target);
+        if (isClickDeleteBtn) {
+            return null;
+        }
+        return onSelect(imageRef);
+    };
+
+    const openModalDelete = async () => {
+        const result = await confirmModal({
+            icon: <TrashIcon />,
+            variant: 'destructive',
+            header: 'Are you sure you want to delete this item?',
+            question: "This item will be deleted immediately. You can't undo this action.",
+            okButtonLabel: 'Delete',
+        });
+
+        if (result) {
+            handleDeleteClick(imageRef);
+        }
+    };
+
     return (
         <StyledFileContainer>
-            <StyledContainerImage type="button" onClick={() => onSelect(imageRef)}>
+            <StyledContainerImage type="button" onClick={handleClick}>
                 <RenderIf isTrue={!!src}>
                     <StyledImage
                         src={src}
                         alt={imageRef.name}
                         onLoad={() => setLoading(false)}
                         $loading={loading}
                     />
+                    <RenderIf isTrue={allowDelete}>
+                        <StyledDeleteButton>
+                            <ButtonIcon
+                                variant="inverse"
+                                size="medium"
+                                icon={<Trash />}
+                                onClick={openModalDelete}
+                                ref={deleteBtn}
+                            />
+                        </StyledDeleteButton>
+                    </RenderIf>
+                    <StyledCheckContainer>
```

Comment:
same ^^ when possible we need to avoid extra wrappers, we can pass the icon through a styled and apply necessary styles